### PR TITLE
[Improve][SeaTunnel-Schema] Support parse row type from config file

### LINKED ISSUE #2765: [Improve][SeaTunnel-Schema] Support parse row type from config file
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

This is a subtask of #2752.

### Usage Scenario

_No response_

### Related issues

#2752 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #2765

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/schema/SeaTunnelSchema.java [hunk 1]
```diff
@@ -189,17 +207,25 @@ private static Map<String, String> convertConfig2Map(Config config) {
...
+            if (value.getNodeType() == JsonNodeType.OBJECT) {
+                fieldsMap.put(key, value.toString());
+            } else {
+                fieldsMap.put(key, value.textValue());
+            }
+        });
+        return fieldsMap;
+    }
+
+    private static SeaTunnelRowType map2SeaTunnelRowType(Map<String, String> fieldsMap) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/schema/SeaTunnelSchema.java]:
```suggestion
    private static SeaTunnelRowType mapToSeaTunnelRowType(Map<String, String> fieldsMap) {
```

File: seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/schema/SeaTunnelSchema.java [hunk 2]
```diff
@@ -189,17 +207,25 @@ private static Map<String, String> convertConfig2Map(Config config) {
...
 
-    public static SeaTunnelSchema buildWithConfig(Config schemaConfig) {
-        CheckResult checkResult = CheckConfigUtil.checkAllExists(schemaConfig, FIELD_KEY);
-        if (!checkResult.isSuccess()) {
-            String errorMsg = String.format("Schema config need option [%s], please correct your config first", FIELD_KEY);
-            throw new RuntimeException(errorMsg);
-        }
-        Config fields = schemaConfig.getConfig(FIELD_KEY);
-        Map<String, String> fieldsMap = convertConfig2Map(fields);
+    private static Map<String, String> convertJson2Map(String json) {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/schema/SeaTunnelSchema.java]:
```suggestion
    private static Map<String, String> convertJsonToMap(String json) {
```

**Review by ashulin — COMMENTED** [review]
Should we support nested Rows? SeaTunnelRow has other information, such as Rowkind, rather than just structured information.

**Comment by hailin0** [comment]
> Should we support nested Rows? SeaTunnelRow has other information, such as Rowkind, rather than just structured information.


This should be the consensus?

**Comment by TyrantLucifer** [comment]
> Should we support nested Rows? SeaTunnelRow has other information, such as Rowkind, rather than just structured information.

In the long run, I think it is necessary to support this feature, after all, there will be some CDC connectors will be used to. Now `SeaTunnelRow` does not have other information, if we want to support this feature we should upgrade `SeaTunnelRow` or add a new data type first.

**Comment by EricJoy2048** [comment]
> > Should we support nested Rows? SeaTunnelRow has other information, such as Rowkind, rather than just structured information.
> 
> In the long run, I think it is necessary to support this feature, after all, there will be some CDC connectors will be used to. Now `SeaTunnelRow` does not have other information, if we want to support this feature we should upgrade `SeaTunnelRow` or add a new data type first.

Yes, I agree with you.

**Comment by ashulin** [comment]
@TyrantLucifer @hailin0 @EricJoy2048 My idea is to add a structured type, and ROW is a special structured type;

**Comment by TyrantLucifer** [comment]
> @TyrantLucifer @hailin0 @EricJoy2048 My idea is to add a structured type, and ROW is a special structured type;

+1

**Comment by EricJoy2048** [comment]
So, How do we deal with this pr? @TyrantLucifer  @Hisoka-X  @ashulin  @hailin0

**Comment by Hisoka-X** [comment]
> So, How do we deal with this pr? @TyrantLucifer @Hisoka-X @ashulin @hailin0

Based on your communication, I think this PR should remove nest type first. And add it after we support nest type in SeaTunnelRow. The other row type can be merged.

**Comment by TyrantLucifer** [comment]
> > So, How do we deal with this pr? @TyrantLucifer @Hisoka-X @ashulin @hailin0
> 
> Based on your communication, I think this PR should remove nest type first. And add it after we support nest type in SeaTunnelRow. The other row type can be merged.

So does the current code merged in? Nest type is not developed completed, we can update this part of code until nest type developed completed. Because this part code can offer the ability of parsing `SeaTunnelRow` type in config file, it very useful for FakeSource generate data.

**Review by Hisoka-X — APPROVED** [review]

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**